pkg/agentmap/generator.go

```diff
@@ -59,10 +59,16 @@ func Generate(ctx context.Context, wl k8sapi.Workload, cfg *GeneratorConfig) (*a
 	}
 
 	var ccs []*agentconfig.Container
-	agentPort := cfg.AgentPort
+	pns := make(map[int32]struct{})
+	portNumber := func(cnPort int32) uint16 {
+		// Increase the agent's port number only for unique container ports
+		pns[cnPort] = struct{}{}
+		return cfg.AgentPort + uint16(len(pns)-1)
```

Comment:
If we are reusing ports, do we always want to reuse the last port?

Reply:
That's not what's happening. The port starts at some given value. Then, for each unique container port that is added, the port increases its value. So if the initial agent port is 9000 and the container ports are 80, 8080, 8080, 443, the the corresponding mappings will be:
```
80 -> 9000
8080 -> 9001
8080 -> 9001 // Mapping is unchanged because len(pns) is unchanged
443 -> 9002
```

Reply:
right. could the order be 80, 8080, 443, 8080? in which case

80 -> 9000
8080 -> 9001
443 -> 9002 // Mapping is unchanged because len(pns) is unchanged
8080 -> 9002

Reply:
Doh, you're right! I need to fix that.